packages/fx-core/resource/yaml-schema.json

```diff
@@ -0,0 +1,1495 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "projectId": {
+      "type": "string",
+      "description": "projectId used for telemetry"
+    },
+    "environmentFolderPath": {
+      "type": "string",
+      "description": "path to the environment folder"
+    },
+    "version": {
+      "type": "string",
+      "description": "The version of the yaml file schema"
+    },
+    "provision": {
+      "$ref": "#/definitions/lifeCycleArray",
+      "description": "Triggered by `teamsfx provision`"
+    },
+    "deploy": {
+      "$ref": "#/definitions/lifeCycleArray",
+      "description": "Triggered by `teamsfx deploy`"
+    },
+    "publish": {
+      "$ref": "#/definitions/lifeCycleArray",
+      "description": "Triggered by `teamsfx publish`"
+    }
+  },
+  "required": ["version"],
+  "definitions": {
+    "lifeCycleArray": {
+      "type": "array",
+      "items": {
+        "anyOf": [
+          { "$ref": "#/definitions/aadAppCreate" },
+          { "$ref": "#/definitions/aadAppUpdate" },
+          { "$ref": "#/definitions/armDeploy" },
+          { "$ref": "#/definitions/azureStorageEnableStaticWebsite" },
+          { "$ref": "#/definitions/cliRunNpmCommand" },
+          { "$ref": "#/definitions/cliRunDotnetCommand" },
+          { "$ref": "#/definitions/cliRunNpxCommand" },
+          { "$ref": "#/definitions/azureStorageDeploy" },
+          { "$ref": "#/definitions/azureAppServiceDeploy" },
+          { "$ref": "#/definitions/azureFunctionsDeploy" },
+          { "$ref": "#/definitions/teamsAppCreate" },
+          { "$ref": "#/definitions/teamsAppValidateManifest" },
+          { "$ref": "#/definitions/teamsAppValidateAppPackage" },
+          { "$ref": "#/definitions/teamsAppZipAppPackage" },
+          { "$ref": "#/definitions/teamsAppUpdate" },
+          { "$ref": "#/definitions/teamsAppPublishAppPackage" },
+          { "$ref": "#/definitions/botAadAppCreate" },
+          { "$ref": "#/definitions/botframeworkCreate" },
+          { "$ref": "#/definitions/fileCreateOrUpdateEnvironmentFile" },
+          { "$ref": "#/definitions/fileCreateOrUpdateJsonFile" },
+          { "$ref": "#/definitions/fileUpdateEnv" },
+          { "$ref": "#/definitions/fileUpdateJson" },
+          { "$ref": "#/definitions/prerequisiteInstall" },
+          { "$ref": "#/definitions/m365TitleAcquire" },
+          { "$ref": "#/definitions/spfxDeploy" },
+          { "$ref": "#/definitions/teamsAppCopyAppPackageToSPFx" },
+          { "$ref": "#/definitions/script" }
+        ]
+      }
+    },
+    "aadAppCreateBase": {
+      "type": "object",
+      "description": "Create Azure Active Directory application and client secret (optional). Please refer to https://aka.ms/teamsfx-actions/aadapp-create for more details.",
+      "required": ["uses", "writeToEnvironmentFile"],
+      "properties": {
+        "name": {
```

Comment:
How do we use this? I see its optional and never generated in our templates ... should we just remove this?

Reply:
Some templates/samples uses "name" field to give each `cli/npmRunCommand` a human readable name. It's usesful when there're more than one `cli/npmRunCommand` actions defined for difference purposes. For example, one is for building tab another is for building bot. Without a human readable name, a user cannot easily tell them apart in both yaml file and summary info.

Reply:
An example is here https://github.com/OfficeDev/TeamsFx-Samples/blob/dev/hello-world-teams-tab-and-outlook-add-in/teamsapp.yml#L52

Reply:
@timngmsft I'm resolving this conversation. Please let me know if there's further questions.